bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -255,10 +255,15 @@ func (r *KubeadmConfigReconciler) handleClusterNotInitialized(ctx context.Contex
 		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
 	}
 
+	machine := &clusterv1.Machine{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(scope.ConfigOwner.Object, machine); err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "cannot convert object to %s", scope.ConfigOwner.GetKind())
```

Comment:
Wouldn't this error message return "cannot convert object to MachinePool" if a MachinePool resource makes it to here?

Reply:
good catch, thanks :)